config/config.go

```diff
@@ -174,3 +172,22 @@ func GetDuration(key string) time.Duration {
 func GetBool(key string) bool {
 	return vip.GetBool(key)
 }
+
+// SubWithDefault returns sub-config by keys including configuration defaults
+// values. It returns nil if no such key. It is analog of the viper.Sub()
+// function. This is workaround for the issue
+// https://github.com/spf13/viper/issues/559
+func SubWithDefault(config *viper.Viper, key string) *viper.Viper {
+	var allSettingsByKey, ok = config.AllSettings()[strings.ToLower(key)]
```

Comment:
Is it possible that a key with upper case characters were added to the config?

Reply:
viper.Viper keeps all keys in low case internally, so it doesn't matter which case was used in key to put the value, when you get all the key/value map back all keys are in low case. For any Get() operation viper.Viper converts key to low case as well.